keyboards/decent/numpad/numpad.c

```diff
@@ -0,0 +1,105 @@
+// Copyright 2024 B. Le Roy
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "quantum.h"
+#include "analog.h"
+
+joystick_config_t joystick_axes[JOYSTICK_AXIS_COUNT] = {
+    [0] = JOYSTICK_AXIS_VIRTUAL,
+    [1] = JOYSTICK_AXIS_VIRTUAL,
+    [2] = JOYSTICK_AXIS_VIRTUAL,
+    [3] = JOYSTICK_AXIS_VIRTUAL
+};
+
+void keyboard_post_init_kb(void) {
+    gpio_set_pin_input_high(GP10); // Up1
+    gpio_set_pin_input_high(GP11); // Down1
+    gpio_set_pin_input_high(GP12); // Left1
+    gpio_set_pin_input_high(GP13); // Right1
+    gpio_set_pin_input_high(GP14); // Fire1
+    gpio_set_pin_input_high(GP15);
+    gpio_set_pin_input_high(GP16);
+    gpio_set_pin_input_high(GP17);
+    gpio_set_pin_input_high(GP18);
+    gpio_set_pin_input_high(GP19);
+}
+
+void set_button_state(int btn, bool state) {
+    if (state) {
+        unregister_joystick_button(btn);
+    }
+    else {
+        register_joystick_button(btn);
+    }
+}
+
+void matrix_scan_kb(void) {
```

Comment:
Probably more appropriate to use [housekeeping](https://docs.qmk.fm/custom_quantum_functions#keyboard-housekeeping) here.